### ISSUE TITLE
Black image on B&W converted images after 3.6.0 update

### ISSUE THREAD

**Issue by WillyKirsch** [darktable-org/darktable#9466]
Images that was converted to B&W in color calibration module on version 3.4 using Ilford HP5+ preset got black in the darkroom view when opened in version 3.6.

Thumbnail keeps visible until I went back to Ligtable view, then got black too.

The problem its solved re-applying the HP5+ preset.

Exported JPGs are also a black rectangle.

Darktable version 3.6.0 running on Windows 10

Follows an screenshot:

![ScreenShot](https://user-images.githubusercontent.com/9601026/125012514-b5f20d00-e040-11eb-904f-b9df053790d8.jpg)

**Comment by kofa73** [comment]
The new preset has the 'clip negative RGB from gamut' option checked; maybe the old one didn't? Negative RGB values are known to cause issues, especially with OpenCL.
![image](https://user-images.githubusercontent.com/3177739/125157302-87705100-e16a-11eb-84d2-0b0fd9af048f.png)

**Comment by johnny-bit** [comment]
This is a candidate for "known issues" I think. Not much to do here because the BW presets do cause negatives to appear and thus cause all sorts of issues and that includes black images.

Presets themselves are fixed, however the images will need that slight adjustments.

**Comment by WillyKirsch** [comment]
> The new preset has the 'clip negative RGB from gamut' option checked; maybe the old one didn't? Negative RGB values are known to cause issues, especially with OpenCL.
> ![image](https://user-images.githubusercontent.com/3177739/125157302-87705100-e16a-11eb-84d2-0b0fd9af048f.png)

It was checked (the default as far as I know) in any case I didn't changed the setting at all in both versions, now I hace it checked in the example photo.

I will try to reproduce the problem since I rolled back to 3.4.1 due to crashes in other modules too. and needed to work with my photos...

I will make an alternate installation to have both running and make some tests...

**Comment by WillyKirsch** [comment]
> This is a candidate for "known issues" I think. Not much to do here because the BW presets do cause negatives to appear and thus cause all sorts of issues and that includes black images.
> 
> Presets themselves are fixed, however the images will need that slight adjustments.

I don't think so, having all my library in B&W corrupted just because a program version update its not trivial for me..

Just to clarify, I don't even make any adjustment in V3.6.0.1 just opened the photo (with the preset applied in the previous version) in the darkroom... and got the black square..

And yes, I'm Using OpenCL

**Comment by WillyKirsch** [comment]
Hi, I'm running both versions in the same PC, 3.4.1.1 and 3.6.0 and on the latest I get black images (as in my first post) just after switching form library to darkroom view (no editing its done on v3.6)... then I can go back and forth and keeps black forever including the thumbnail.

I close V3.6.0 and open the same raw (and xmp) file in 3.4.1.1 and all its fine, so I confirm, the problem is in the new version, the files are not corrupted, using 3.4 version on the same file it appear perfect as always...

I can share any data or run a debugging command to collect data if you want and tell me what to do.

**Comment by kofa73** [comment]
The trouble is, if error handling (for the negative values) is added to the calculations, they'll become slow. Have you tried toggling the 'clip negative RGB from gamut' flag?

**Comment by LuisGAbarca-DT** [comment]
Hi there, I am having the exact same problem, even with color photos. A black square when exporting, or a black square when in Lighttable. I am using Windows 10, laptop Lenovo, AMD Ryzen 5, AMD Radeon. This is frustrating I am going back to version 3.4.

**Comment by kofa73** [comment]
@LuisGAbarca-DT : have you tried the suggested solution?
https://github.com/darktable-org/darktable/issues/9466#issuecomment-878805445

**Comment by kofa73** [comment]
@WillyKirsch : I've just checked your history stack. I see you use the base curve, so the legacy pipeline. I think it may not be getting much attention now, as it's considered legacy. Same is the case with 'basic adjustments'.
Also, could you share an example raw + xmp?

**Comment by LuisGAbarca-DT** [comment]
Hi kofa73, I checked the above solution and it did not work. I am having the problem not only with B&W also color.Actually when in lightroom some of the thumbs are pixelated or completely black. Others if they look as they should, then using the W key they go black, others black when using the W key I can see the photo as it should for one second,, then all black...very strange...What do you need me to send you to check the problem? Thanks.

**Comment by LuisGAbarca-DT** [comment]
Something else, not all the photos behave this way...

**Comment by kofa73** [comment]
Providing the information requested in the bug report template would be a good start... you can open it here:
https://github.com/darktable-org/darktable/issues/new?assignees=&labels=&template=bug_report.md&title=

Darktable version, OS version, video card type, driver version, a raw file + XMP pair exhibiting the problem (as already requested) be great.
Also, if you can run darktable from the command-line, using
`darktable -d opencl` might reveal something (though the developers may suggest more or different logging options)

**Comment by WillyKirsch** [comment]
> @WillyKirsch : I've just checked your history stack. I see you use the base curve, so the legacy pipeline. I think it may not be getting much attention now, as it's considered legacy. Same is the case with 'basic adjustments'.
> Also, could you share an example raw + xmp?

Hey thank you! yes, let me get one example and I'll post it!

I understand that Legacy modules get less attention, my real issues is: I have to "re-processes" my legacy catalog due to an update (because I used deprecated modules)? its fine if I have to let go old modules, but I don't think its a good idea to have to re process old photos... Actually, I'm still working with 3.4 version... I'm not confident with v3.6...

Regards!!

**Comment by WillyKirsch** [comment]
> @WillyKirsch : I've just checked your history stack. I see you use the base curve, so the legacy pipeline. I think it may not be getting much attention now, as it's considered legacy. Same is the case with 'basic adjustments'.
> Also, could you share an example raw + xmp?

Hi, I'm attaching a CR2 and xmp files for a photo suffering the issue, if you go back to the step 12 in the HS before to apply the HP5+ preset in the CC module, you recover the image. There is still the BC module.

Thank you!

[IMG_2276.CR2.zip](https://github.com/darktable-org/darktable/files/6873904/IMG_2276.CR2.zip)

**Comment by kofa73** [comment]
Works fine with the current development build (master).
![image](https://user-images.githubusercontent.com/3177739/127047671-3f0b60d9-817e-45e8-abd8-9109084da1cf.png)

**Comment by WillyKirsch** [comment]
> I don't Know what to say... Follows my setup:
> 
> Win 10Pro 20H2 compilation 19042.1110
> Nvidia GTX 750Ti Driver version 456.71
> 
> I'll check updating the NVIDIA drivers

I updated to 471.41 and nothing changed...

**Comment by kofa73** [comment]
Have you tried `darktable -d opencl`?
Also, maybe the issue has been fixed -- can you try making a copy of the image and XMP and run the build published here?
https://discuss.pixls.us/t/darktable-windows-insider-program-7-25/26087
(Just to be safe, back up your DB -- I think this release uses alternative database and config paths, but better safe than sorry.)
You can read more about these builds here: https://discuss.pixls.us/t/darktable-windows-insider-program/25740

Can you check how much GPU memory is in use? Under Linux, it's done with the command-line nvdia-smi tool:
```
ofa@eagle:~$ nvidia-smi
Thu Jul 29 08:55:38 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:09:00.0  On |                  N/A |
|  0%   39C    P8     9W / 120W |    300MiB /  6077MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```
So, in my case, it's 300 MB. Reserve at least this much in your config file using `opencl_memory_headroom`. A higher number will block more memory from darktable, but I think one needs to tell dt how much is needed for the OS and other programs; otherwise it will think that memory is available, and allocation attempts would probably fail.

I guess under Windows you'll have some control GUI (there's one for Linux as well):
![image](https://user-images.githubusercontent.com/3177739/127446699-69cc7637-6475-4bda-af3f-428608989a8a.png)
As you can see, the memory requirement changed somewhat between the two (300 vs 311 MB), so don't take the reported value, add some extra headroom.

**Comment by WillyKirsch** [comment]
Thank you! 

I'm using OpenCL

I will check all your suggestions and come back with news

**Comment by WillyKirsch** [comment]
Well, I made some additional tests and found this:

I checked with old pictures that was converted to B&W with the channel mixer (the pre-CC module era) and all its fine.

The Color calibration module its the problem, I found one picture with just the basic modules applied (including base curve) and the CC module fails.

I will try to develop a picture from scratch no using the Base curve module to check if it is related

in pictures were the CC module its used but not the HP5+ preset the problem does not rise...

In all the cases the problem its solved re applying the HP5+.

I'm attaching the debug log of the OpenCL library and there is just a warning that I have no clue what it means... but all the modules are compiled and loaded with no problems

This is the warning:
(darktable:18160): GLib-GIO-WARNING **: 20:46:41.979: Unexpectedly, UWP app `Microsoft.ZuneVideo_10.21061.10121.0_x64__8wekyb3d8bbwe' (AUMId `Microsoft.ZuneVideo_8wekyb3d8bbwe!Microsoft.ZuneVideo') supports 33 extensions but has no verbs

[darktable-log.txt](https://github.com/darktable-org/darktable/files/6904250/darktable-log.txt)

I have to check the memory allocation matter that I have no checked yet.

I'll have to check the 3.7 compilation to verify what happens...

Thank you!!

**Comment by WillyKirsch** [comment]
Hey!!

In the 3.7 Version, appear to work fine!! 

just one picture got black but had to be a thumbnail matter because the zoom miniature its ok, and the exported picture too...

I went back to 3.4.1 and there was a black image too but all the same in 3.7...

then I went to 3.6 and voila! the black image gone... and can't reproduce it...

About the memory allocation I didn't any test.. I had no time to investigate how to change it and test...

I've checked processing without the base curve module and its the same...

I isolated the issue to the color calibration module, and going back and forth finally I "cleared" all the images and I have no issue in 3.6...

I you need some specific test, I can go to the back up and recover the failing pictures

In conclusion, it appear that the 3.7 version has no issue...

**Comment by WillyKirsch** [comment]
> So, in my case, it's 300 MB. Reserve at least this much in your config file using `opencl_memory_headroom`. A higher number will block more memory from darktable, but I think one needs to tell dt how much is needed for the OS and other programs; otherwise it will think that memory is available, and allocation attempts would probably fail.

I have this setting: opencl_memory_headroom=400

I'm no completely clear about if its better to have more or less headroom...
